src/plugins/event-plugins/ResourcePlugin.ts

```diff
@@ -90,7 +90,17 @@ export class ResourcePlugin extends InternalPlugin {
     }: PerformanceResourceTiming): void => {
         const pathRegex =
             /.*\/application\/[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}\/events/;
-        const entryUrl = new URL(name);
+        let entryUrl: URL;
+        try {
+            entryUrl = new URL(name);
+        } catch {
+            // Throw error if constructing URL for name fails. Log name
+            // value with the error to identify patterns that cause such
+            // failures.
+            throw new Error(
+                `Unable to construct URL from PerformanceResourceTiming name: ${name}`
+            );
+        }
         if (
             entryUrl.host === this.context.config.endpointUrl.host &&
             pathRegex.test(entryUrl.pathname)
```

Comment:
suggestion: Since this is the only place we use `entryUrl`, we could (1) modify this condition to only run when `entryUrl` is truthy, and (2) catch and ignore the error thrown by `new URL(name)`.

There is a [configuration option for filtering PerformanceResourceTiming events](https://github.com/aws-observability/aws-rum-web/blob/main/docs/configuration.md#performance), and by default events are removed when `name` is not a URL with an http protocol. So by ignoring the error, I believe we solve the problem of spurious error messages without losing functionality.

I am ok with merging this PR after making this change and adding a unit test to cover the behavior,

Reply:
@williazz observed that [filtering happens](https://github.com/aws-observability/aws-rum-web/blob/a915864329b7d9b2718c311f1593bf61ee1f82f0/src/plugins/event-plugins/ResourcePlugin.ts#L59) before we reach this point. So @shubhsheth unless you have modified the filter, to reach this condition this URL must begin with `http`, but still be an invalid URL.

Reply:
Thanks @qhanam for looking further into this. I've updated the PR with the suggested changes.

It is quite interesting what @williazz discovered about the filters. I do not see any modification to the filtering in the deployed changes.